app/assets/javascripts/admin/spree/orders/variant_autocomplete.js.erb

```diff
@@ -54,46 +54,53 @@ $(document).ready(function() {
     }
     $('a.save-item').click(handle_save_click);
 
-    handle_delete_click = function(){
-      var del = $(this);
-      var shipment_number = del.data('shipment-number');
-      var variant_id = del.data('variant-id');
+    handle_delete_click = function(e, confirmed){
+      if (confirmed) {
+        var del = $(this);
+        var shipment_number = del.data('shipment-number');
+        var variant_id = del.data('variant-id');
 
-      toggleItemEdit();
+        toggleItemEdit();
 
-      adjustItems(shipment_number, variant_id, 0);
+        adjustItems(shipment_number, variant_id, 0);
+      }
     }
-    $('a.delete-item').click(handle_delete_click);
+    $('a.delete-item').on('confirm:complete', handle_delete_click);
```

Comment:
I trust that this works but I'm curious what's actually happening here - we're now relying on the return value of `adjustItems`, which will be false if we shouldn't remove the last item, is that basically it?

Reply:
This is just to prevent a bug: the alert box that shows up when the user clicks the delete button was not having any effect, since just clicking the button already trigered the `handle_delete_click` callback. Now its triggered by `confirm:complete`, i. e. when the user clicks the confirm button in the alert. If the user clicks the cancel button the function won't be called. This is also described in the issue:

By editing an order, /admin/orders/<order_number>/edit it is possible to remove the last item here it's possible both by clicking the trash-bin (**somehow clicking OK seems not to be required**) and by reducing the number of items to zero.